File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/video/RecordingImpl.kt
package dev.zeekr.camerax_android.video

import androidx.camera.video.Recording
import dev.zeekr.camerax_android.CameraXApiPigeonProxyApiRegistrar
import dev.zeekr.camerax_android.PigeonApiRecordingProxyApi

class RecordingImpl(registrar: CameraXApiPigeonProxyApiRegistrar) : PigeonApiRecordingProxyApi(registrar) {
    @androidx.camera.video.ExperimentalPersistentRecording
    override fun isPersistent(pigeon_instance: Recording): Boolean {
        return pigeon_instance.isPersistent
    }

    override fun mute(pigeon_instance: Recording, muted: Boolean) {
        return pigeon_instance.mute(muted)
    }

    override fun pause(pigeon_instance: Recording) {
        return pigeon_instance.pause()
    }

    override fun resume(pigeon_instance: Recording) {
        return pigeon_instance.resume()
    }

    override fun stop(pigeon_instance: Recording) {
        return pigeon_instance.stop()
    }
}

File: camerax_android/lib/src/video/recording_impl.dart
import 'package:camerax_android/src/camerax_api.g.dart';
import 'package:camerax_android/src/common.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class RecordingImpl extends Recording with AutoCloseableImpl {
  @override
  final RecordingApi api;

  RecordingImpl.internal(this.api) : super.impl();

  @override
  Future<void> mute(bool muted) => api.mute(muted);

  @override
  Future<void> pause() => api.pause();

  @override
  Future<void> resume() => api.resume();

  @override
  Future<void> stop() => api.stop();
}

extension RecordingApiX on RecordingApi {
  RecordingImpl get impl {
    return RecordingImpl.internal(this);
  }
}
